packages/xod-client-electron/src/app/workspaceActions.js

```diff
@@ -171,6 +172,22 @@ export const loadProjectsOrSpawnDefault = R.curry(
   )(workspacePath)
 );
 
+// :: SaveLibPayload :: { request: LibParams, xodball: Project }
+// :: IpcEvent -> SaveLibPayload -> Promise SaveLibPayload Error
```

Comment:
I think `xodball: Project` is a source of confusion. Is it a xodball or project? These two things have a different shape.